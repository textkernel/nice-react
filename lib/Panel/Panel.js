import React from 'react';
import PropTypes from 'prop-types';

const Panel = ({ title, children, color, inverseHeading }) => {
  const headerDiv = title ? <div className='nice-panel-heading'>{title}</div> : '';

  return (
    <div className={`nice-panel ${color}`}>
      {
        inverseHeading === false &&
          headerDiv
      }

      <div className='nice-panel-content'> 
        {children}
      </div>

      {
        inverseHeading &&
          headerDiv
      }
      
    </div>
  );
};

Panel.propTypes = {
  title: PropTypes.string,
  color: PropTypes.string,
  inverseHeading: PropTypes.bool,
  children: PropTypes.node
};

Panel.defaultProps = {
  title: null,
  color: '',
  inverseHeading: false,
  children: null
};  


export default Panel;