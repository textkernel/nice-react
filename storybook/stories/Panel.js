import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Panel } from 'nice-react';

storiesOf('Panel', module)
  .add('empty', () => (
    <Panel> This is a panel without heading </Panel>
  ))
  .add('with title', () => (
    <Panel title={'My Panel'}> This panel has a heading </Panel>
  ))
  .add('with brand color', () => (
    <Panel title={'Very well'} color={'good'}> This is a good panel </Panel>
  ))
  .add('with inverse heading', () => (
    <Panel title={'Opss..'} color={'bad'} inverseHeading> This is a bad panel </Panel>
  ));;