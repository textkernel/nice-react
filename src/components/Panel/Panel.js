import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Panel.scss';

const Panel = ({ children, className, context, inverseHeading, style, title }) => {
    const headerDiv = title ? <div className="nice-panel-heading">{ title }</div> : '';

    return (
        <div
            className={ classNames('nice-panel', className, context) }
            style={ style }
        >
            {
                inverseHeading === false &&
                headerDiv
            }

            <div className="nice-panel-content">
                { children }
            </div>

            {
                inverseHeading &&
                headerDiv
            }
            
        </div>
    );
};

Panel.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    context: PropTypes.string,
    inverseHeading: PropTypes.bool,
    style: PropTypes.object,
    title: PropTypes.string
};

Panel.defaultProps = {
    children: null,
    className: '',
    context: '',
    inverseHeading: false,
    style: null,
    title: null
};

export default Panel;
