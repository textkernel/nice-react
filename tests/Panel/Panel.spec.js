import React from 'react';
import Panel from '../../lib/Panel';

describe('<Panel />', () => {
  it('should render content  the Panel', () => {
    const wrapper = shallow(<Panel>Hello Jest!</Panel>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Panel with title', () => {
    const wrapper = shallow(<Panel title='My title' />);
    expect(wrapper.find('.nice-panel-heading').text()).toEqual('My title');
  });

  it('should render the Panel header on the bottom', () => {

  });

  it('should render with different color theme', () => {

  });
});