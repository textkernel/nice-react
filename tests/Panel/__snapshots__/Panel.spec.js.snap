// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> should render content  the Panel 1`] = `
<div
  className="nice-panel "
>
  <div
    className="nice-panel-content"
  >
    Hello Jest!
  </div>
</div>
`;
